staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -631,17 +631,25 @@ func (r *Request) Watch(ctx context.Context) (watch.Interface, error) {
 		}
 
 		done, transformErr := func() (bool, error) {
-			defer readAndCloseResponseBody(resp)
+			closer := NewResponseCloser(resp)
+			defer closer.DrainAndClose()
+
+			// get the result out of the response if provided *before* IsNextRetry,
+			// since that has to drain/close the response before resetting the request body
+			var result Result
+			if resp != nil {
+				result = r.transformResponse(resp, req)
+			}
```

Comment:
I like the idea of buffering the body into a copy in IsNextRetry... if we do that, maybe we can keep the transformResponse call in the original location?